addons/stock/models/stock_orderpoint.py

```diff
@@ -408,8 +408,9 @@ def _get_orderpoint_action(self):
 
         orderpoints = self.env['stock.warehouse.orderpoint'].with_user(SUPERUSER_ID).create(orderpoint_values_list)
         for orderpoint in orderpoints:
-            orderpoint.route_id = orderpoint.product_id.route_ids[:1]
-        orderpoints.filtered(lambda o: not o.route_id)._set_default_route_id()
+            orderpoint_wh = orderpoint.location_id.get_warehouse()
+            orderpoint.route_id = next((r for r in orderpoint.product_id.route_ids if not r.supplied_wh_id or r.supplied_wh_id == orderpoint_wh), None) \
+                                  or orderpoint._set_default_route_id()
```

Comment:
this is wrong. `_set_default_route_id` method doesn't return anything. so in case the _next_ is None, then orderpoint.route_id will be None, as also `orderpoint._set_default_route_id()` returns None.

```suggestion
            orderpoint.route_id = next((r for r in orderpoint.product_id.route_ids if not r.supplied_wh_id or r.supplied_wh_id == orderpoint_wh), orderpoint.route_id)
            if not orderpoint.route_id:
                orderpoint._set_default_route_id()
```